packages/components/modules/messages/ChatRoom/index.tsx

```diff
@@ -32,6 +35,34 @@ const ChatRoom: FC<ChatRoomProps> = ({
       fetchKey: roomId,
     },
   )
+  const { currentProfile } = useCurrentProfile()
+  const [commitMutation] = useReadMessageMutation()
+
+  const prevProfileIdRef = useRef<string | undefined>(currentProfile?.id)
+  const prevRoomIdRef = useRef<string | undefined>(chatRoom?.id)
+
+  useEffect(() => {
+    if (hasRunRef.current)
```

Comment:
Would this break in non-strict mode? But if I understood you right, you will remove this in the next PR anyway, so it does not matter.